test/integration/TrueFiPool2.test.ts

```diff
@@ -19,6 +19,7 @@ describe('TrueFiPool2', () => {
   const USDC_ADDRESS = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'
   const TFUSDT_ADDRESS = '0x6002b1dcb26e7b1aa797a17551c6f487923299d7'
   const TFUSDT_STRATEGY_ADDRESS = '0x8D162Caa649e981E2a0b0ba5908A77f2536B11A8'
+  const PROXY_ADDRESS = '0x0BA5908a77f2536b11a88d162CAa649E981E2a0b'
```

Comment:
I'm not sure I understand this fix. Where does this address come from? I don't see it on Etherscan:
https://etherscan.io/address/0x0BA5908a77f2536b11a88d162CAa649E981E2a0b

Reply:
This is a perfect learning moment, since we also want to write one of these "integration" tests for the new TrueUSDNoDelegate. (You can also see a minimal example in PR #1177, which also should be reviewed + merged at some point -- the action for which we wrote that test has already been queued and executed on the multisig.)

We use Ganache locally to run the test _on a mainnet fork_. We fork mainnet in the `forkChain()` call in line 27. You can optionally specify:
- unlocked account addresses with particular properties (e.g., the `owner()` or `proxyOwner()` of the contract, or ETH_HOLDER is just a mainnet address that has a bunch of tokens), and
- the block number to fork, in case there's some specific past contract state that we want to test.

Reply:
Using the same address is a hacky workaround meant specifically for proxies. A proxy only supports a couple native functions, like `proxyOwner()`, `transferProxyOwnership()`, `upgradeTo()`, etc. Every other selector is forwarded on-chain to the implementation address's ABI.

Unfortunately Ganache and Ethers don't know this, so they complain whenever we issue a tx to a proxy that doesn't match its native ABI.

To work around this limitation when we need to make both upgrade and impl calls in the same test, we can connect to the same proxy address with its proxy ABI and its impl ABI as "separate" contracts.